packages/redux/src/products/reducer/groupingProperties.ts

```diff
@@ -0,0 +1,76 @@
+import * as actionTypes from '../actionTypes';
+import { AnyAction, combineReducers } from 'redux';
+import type {
+  FetchProductGroupingPropertiesAction,
+  FetchProductGroupingPropertiesFailureAction,
+  FetchProductGroupingPropertiesRequestAction,
+  ProductsGroupingPropertiesState,
+} from '../types';
+
+export const INITIAL_STATE: ProductsGroupingPropertiesState = {
+  error: {},
+  isLoading: {},
+};
+
+const error = (
+  state = INITIAL_STATE.error,
+  action:
+    | FetchProductGroupingPropertiesRequestAction
+    | FetchProductGroupingPropertiesFailureAction,
+) => {
+  switch (action.type) {
+    case actionTypes.FETCH_PRODUCT_GROUPING_PROPERTIES_REQUEST:
+      return {
+        ...state,
+        [action.meta.productId]: undefined,
+      };
+    case actionTypes.FETCH_PRODUCT_GROUPING_PROPERTIES_FAILURE:
+      return {
+        ...state,
+        [action.meta.productId]: action.payload.error,
+      };
+    default:
+      return state;
+  }
+};
+
+const isLoading = (
+  state = INITIAL_STATE.isLoading,
+  action: FetchProductGroupingPropertiesAction | AnyAction,
+) => {
+  switch (action.type) {
+    case actionTypes.FETCH_PRODUCT_GROUPING_PROPERTIES_REQUEST:
+      return {
+        ...state,
+        [action.meta.productId]: true,
+      };
+    case actionTypes.FETCH_PRODUCT_GROUPING_PROPERTIES_SUCCESS:
+      return { ...state, [action.meta.productId]: false };
+    case actionTypes.FETCH_PRODUCT_GROUPING_PROPERTIES_FAILURE:
+      return { ...state, [action.meta.productId]: undefined };
```

Comment:
Is the loading state supposed to be undefined on failure? This will make the `isProductGroupingLoading` selector return not only a boolean value, even after a request has been made 🤔 

(Seeing another reducer in a recent PR, [the `isLoading` is false on failure](https://github.com/Farfetch/blackout/blob/50455210f56c06007cd6a303e397660aa634078a/packages/redux/src/search/reducer/searchSuggestions.ts#L24))

If it's supposed, can you please explain the reasoning to have different approaches for me (and anyone who sees it) to understand? 🙏

Reply:
You're right, it's not supposed to be undefined on failure. I based it on the grouping reducer. I'll fix it here and there as well